src/nvim/os/stdpaths.c

```diff
@@ -0,0 +1,100 @@
+#include "nvim/os/os.h"
+#include "nvim/strings.h"
+#include "nvim/path.h"
+#include "nvim/garray.h"
+
+typedef enum {
+  kXDGConfigHome,
+  kXDGDataHome,
+  kXDGCacheHome,
+  kXDGRuntimeDir,
+  kXDGConfigDirs,
+  kXDGDataDirs,
+} XDGDirType;
+
+static const char *xdg_env_vars[] = {
+  [kXDGConfigHome] = "XDG_CONFIG_HOME",
+  [kXDGDataHome] = "XDG_DATA_HOME",
+  [kXDGCacheHome] = "XDG_CACHE_HOME",
+  [kXDGRuntimeDir] = "XDG_RUNTIME_DIR",
+  [kXDGConfigDirs] = "XDG_CONFIG_DIRS",
+  [kXDGDataDirs] = "XDG_DATA_DIRS",
+};
+
+static const char *const xdg_defaults[] = {
+  // Windows, Apple stuff are just shims right now
+#ifdef WIN32
+  // Windows
+#elif APPLE
```

Comment:
Will there be a way to get neovim on OS X to use XDG basedirs?

OS X behaving differently to other *nixes will be irritating for keeping dotfiles synced.

That and ~/Library is pretty much only used by GUI apps; all the terminal stuff uses dotfiles in ~ or XDG.